lib/balanced_vrp_clustering.rb

```diff
@@ -134,15 +187,96 @@ def build(data_set, cut_symbol, cut_ratio = 1.0, options = {})
         self
       end
 
+      def move_limit_violating_dataitems
+        @needed_moving = @items_with_limit_violation.size
+        mean_distance_diff = @items_with_limit_violation.collect{ |d| d[1] }.mean
+        mean_ratio = @items_with_limit_violation.collect{ |d| d[2] }.mean
+
+        @moved_down = 0
+        @moved_up = 0
+
+        # TODO: check if any other type of ordering might help
+        # Since the last one that is moved up will appear at the very top and vice-versa for the down.
+        # We need the most distant ones to closer to the top so that the cluster can move to that direction next iteration
+        # but a random order might work better for the down ones since they are in the middle and the border is not a straight line
+        # nothing vanilla order 9/34
+        # @items_with_limit_violation.shuffle!  7/34 not much of help. it increases normal iteration count but decreases the loop time
+        # @items_with_limit_violation.sort_by!{ |i| i[5] } 6/34 fails .. good
+        # @items_with_limit_violation.sort_by!{ |i| -i[5] } 5/34 fails .. better
+        # 0.33sort+ and 0.66sort-  #8/21 fails ... bad
+        # 0.33shuffle and 0.66sort- #7/20 fails ... bad
+
+        @items_with_limit_violation.sort_by!{ |i| -i[5] }
```

Comment:
move this line inside `until @items_with_limit_violation.empty? do` to avoid a `if`

Reply:
If we do that, it will get sorted multiple times --  `until condition do/end` is a while loop. 

We don't really need an `if` here anyways, sort_by handles empty arrays correctly by default.